### PR TITLE
Reader of CrystalMap from Bruker's HDF5 file, rewrite EMsoft reader

### PR DESCRIPTION
**Pull request by hakonanes**
Signed-off-by: Håkon Wiik Ånes <hwaanes@gmail.com>

#### Description of the change
* Reader of a crystal map from Bruker's HDF5 file (h5ebsd format) from orientation data obtained from Hough indexing of either EBSD or Transmission Kikuchi Diffraction (TKD) data.
* Since this format is similar to EMsoft's HDF5 file (h5ebsd format), I rewrote this reader as well. Both h5ebsd readers are now implemented as classes inheriting from a base `H5ebsdFile` class. This will make it easier to add a EDAX TSL h5ebsd reader and Oxford Instruments h5ebsd reader in the future.

#### Progress of the PR
- [x] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [x] Unit tests with pytest for all lines
- [x] Clean code style by [running black via pre-commit](https://orix.readthedocs.io/en/latest/contributing.html#code-style)

#### Minimal example of the bug fix or new feature
```python
>>> from orix import io
>>> xmap = io.load("file.h5")
>>> xmap
Phase   Orientations         Name  Space group  Point group  Proper point group     Color
   -1   1178 (23.4%)  not_indexed         None         None                None         w
    1   3862 (76.6%)         Gold        Fm-3m         m-3m                 432  tab:blue
Properties: PCX, PCY, DD, MAD, NIndexedBands, RadonBandCount, RadonQuality, XBEAM, YBEAM, XSAMPLE, YSAMPLE, ZSAMPLE
Scan unit: um
```

#### For reviewers

- [x] The PR title is short, concise, and will make sense 1 year later.
- [ ] New functions are imported in corresponding `__init__.py`.
- [ ] New features, API changes, and deprecations are mentioned in the
      unreleased section in `CHANGELOG.rst`.

### CODE REVIEW

File: orix/io/plugins/_h5ebsd.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+    def close(self):
+        """Close the HDF5 file."""
+        self.file.close()
+
+    def get_dictionary(self, group_name, **kwargs):
+        """Return a dictionary from a data set group.
+
+        Parameters
+        ----------
```
Comment (dnjohnstone) [inline orix/io/plugins/_h5ebsd.py:110]: Please could you have another quick look at this docstring? looks a little incomplete on what group_name and dict are?

File: orix/tests/io/test_io.py [hunk 2]
```diff
@@ -88,14 +88,20 @@ def test_load_unsupported_format(self, temp_file_path):
             _ = load(temp_file_path)
 
     @pytest.mark.parametrize(
-        "top_group, expected_plugin",
-        [("Scan 1", emsoft_h5ebsd), ("crystal_map", orix_hdf5), ("Scan 2", None)],
+        "manufacturer, expected_plugin",
+        [
+            ("EMEBSDDictionaryIndexing.f90", emsoft_h5ebsd),
+            ("orix", orix_hdf5),
+            ("Oxford", None),
```
Comment (dnjohnstone) [inline orix/tests/io/test_io.py:95]: Do you want to test Bruker here also now?

**Review by dnjohnstone — CHANGES_REQUESTED** [review]
This basically looks good to me - a couple of very minor possible things flagged

File: orix/io/plugins/_h5ebsd.py (hunk 1 above)
Comment (hakonanes) [reply orix/io/plugins/_h5ebsd.py:111]: I've updated the docstring with a clear reference for a developer to look at the referenced function for further details on the parameters. This class will always be private, so I think it's OK to not explain its use in great detail in this method.

File: orix/tests/io/test_io.py (hunk 2 above)
Comment (hakonanes) [reply orix/tests/io/test_io.py:94]: Yes, and I've updated the test to do so now. Thanks!

**Review by pc494 — APPROVED** [review]
This LGTM, merging

**Comment by hakonanes** [comment]
Great, thanks!

**Pull request merged**